src/gleam/erlang/process.gleam

```diff
@@ -522,172 +464,143 @@ pub type ProcessDown {
+pub fn deselecting_specific_monitor(
   selector: Selector(payload),
-  monitor: ProcessMonitor,
+  monitor: Monitor,
 ) -> Selector(payload) {
-  remove_selector_handler(selector, monitor.tag)
+  remove_selector_handler(selector, monitor)
 }
 
-// This function is based off of Erlang's gen:do_call/4.
-/// Send a message to a process and wait for a reply.
-///
-/// If the receiving process exits or does not reply within the allowed amount
-/// of time then an error is returned.
-///
-/// The `within` parameter specifies the timeout duration in milliseconds.
-///
-pub fn try_call(
-  subject: Subject(request),
-  make_request: fn(Subject(response)) -> request,
-  within timeout: Int,
-) -> Result(response, CallError(response)) {
+fn perform_call(
+  subject: Subject(message),
+  make_request: fn(Subject(reply)) -> message,
+  run_selector: fn(Selector(reply)) -> Result(reply, Nil),
+) -> reply {
   let reply_subject = new_subject()
+  let assert Ok(callee) = subject_owner(subject)
+    as "Callee subject had no owner"
 
   // Monitor the callee process so we can tell if it goes down (meaning we
   // won't get a reply)
-  let monitor = monitor_process(subject_owner(subject))
+  let monitor = monitor(callee)
 
   // Send the request to the process over the channel
   send(subject, make_request(reply_subject))
 
   // Await a reply or handle failure modes (timeout, process down, etc)
-  let result =
+  let reply =
     new_selector()
-    |> selecting(reply_subject, Ok)
-    |> selecting_process_down(monitor, fn(down: ProcessDown) {
-      Error(CalleeDown(reason: down.reason))
+    |> selecting(reply_subject, function.identity)
+    |> selecting_specific_monitor(monitor, fn(down) {
+      panic as { "callee exited: " <> string.inspect(down) }
     })
-    |> select(timeout)
+    |> run_selector
+
+  let assert Ok(reply) = reply as "callee did not send reply before timeout"
 
   // Demonitor the process and close the channels as we're done
   demonitor_process(monitor)
 
-  // Prepare an appropriate error (if present) for the caller
-  case result {
-    Error(Nil) -> Error(CallTimeout)
-    Ok(res) -> res
-  }
+  reply
 }
 
-/// Send a message to a process and wait for a reply.
-///
-/// If the receiving process exits or does not reply within the allowed amount
-/// of time the calling process crashes. If you wish an error to be returned
-/// instead see the `try_call` function.
+// This function is based off of Erlang's gen:do_call/4.
+/// Send a message to a process and wait a given number of milliseconds for a
+/// reply.
 ///
-/// The `within` parameter specifies the timeout duration in milliseconds.
+/// This function will panic under the following circumstances:
+/// - The callee process exited prior to sending a reply.
+/// - The callee process did not send a reply within the permitted amount of
+///   time.
+/// - The subject is a named subject but no process is registered with that
+///   name.
 ///
 pub fn call(
```

Comment:
Just by reading the type and doc it's a bit hard for me to understand how the two args are used, an example here would be really useful I think